src/ispc/kernels/lanczos3.ispc

```diff
@@ -68,44 +93,48 @@ static inline uint8<4> resample_internal(uniform Image src_image, float<2> uv, u
 
             int addr = (src_kernel_coord.x + src_kernel_coord.y * src_image.size.x) * num_channels;
 
-            float<4> texel;
-
-            const float inv_255 = rcp(255.0);
-
-            if (num_channels == 3) {
-                texel.x = byte_to_float(src_image.data[addr + 0]) * inv_255;
-                texel.y = byte_to_float(src_image.data[addr + 1]) * inv_255;
-                texel.z = byte_to_float(src_image.data[addr + 2]) * inv_255;
-            } else if (num_channels == 4) {
-                texel.x = byte_to_float(src_image.data[addr + 0]) * inv_255;
-                texel.y = byte_to_float(src_image.data[addr + 1]) * inv_255;
-                texel.z = byte_to_float(src_image.data[addr + 2]) * inv_255;
-                texel.w = byte_to_float(src_image.data[addr + 3]) * inv_255;
-            }
+            for (uniform int i = 0; i < num_channels; i++)
+                col[i] += w * byte_to_float(src_image.data[addr + i]);
```

Comment:
Any reason you undid this loop unroll? Iirc we gained quite a bit of performance through it.

Reply:
Trying this plainly on top of #38, which is at:

    Downsample `square_test.png` using ispc_downsampler
                        time:   [29.893 ms 29.922 ms 29.951 ms]

Changing this to:

```ispc
            for (uniform uint8 i = 0; i < num_channels; ++i)
                col[i] += w * byte_to_float(src_image.data[addr + i]) * inv_255;
```

Results in:

    Downsample `square_test.png` using ispc_downsampler
                        time:   [30.943 ms 30.999 ms 31.062 ms]
                        change: [+3.3825% +3.5980% +3.8422%] (p = 0.00 < 0.05)
                        Performance has regressed.

Reply:
Unrolling the loop in `export void resample()`:

https://github.com/Traverse-Research/ispc-downsampler/blob/c21b5dbf70d7d782ea68e99e3b5c3fb004968a02/src/ispc/kernels/lanczos3.ispc#L105-L106

With:

```cpp
        unsigned int addr = (x + y * target_width) * num_channels;

        if (num_channels == 3) {
            out_data[addr + 0] = s.x;
            out_data[addr + 1] = s.y;
            out_data[addr + 2] = s.z;
        } else if (num_channels == 4) {
            out_data[addr + 0] = s.x;
            out_data[addr + 1] = s.y;
            out_data[addr + 2] = s.z;
            out_data[addr + 3] = s.w;
        }
```

Is similarly a good idea:

    Downsample `square_test.png` using ispc_downsampler
                        time:   [29.142 ms 29.169 ms 29.197 ms]
                        change: [-2.1635% -1.8557% -1.6349%] (p = 0.00 < 0.05)
                        Performance has improved.

Reply:
I went ahead to chase the template-parameter version:

```
warning: (ISPC) src/ispc/kernels/lanczos3.ispc:43:10: Error: syntax error, unexpected
warning: (ISPC)         'unsigned', expecting 'typename' or 'int' or '>'.
warning: (ISPC) template<unsigned int NC>
warning: (ISPC)          ^^^^^^^^
```

```
warning: (ISPC) src/ispc/kernels/lanczos3.ispc:43:10: Error: Non-type template parameters are
warning: (ISPC)         not yet supported.
warning: (ISPC) template<int NC>
warning: (ISPC)          ^^^^^^
```

On the compiler built from today's git release (`ispc --version` points to https://github.com/ispc/ispc/commit/5725fb8d495980af52b417ae0c7a89e2a258d2bd).